server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -150,56 +150,56 @@ private void extensionsDiscovery() throws IOException {
         if (!FileSystemUtils.isAccessibleDirectory(extensionsPath, logger)) {
             return;
         }
-        List<Path> pluginDirs = PluginsService.findPluginDirs(extensionsPath);
 
-        Set<Extension> extensions = new HashSet<Extension>();
-        if (!pluginDirs.isEmpty()) {
+        List<Extension> extensions = new ArrayList<Extension>();
+        if (Files.exists(extensionsPath.resolve("extensions.yml"))) {
             try {
                 extensions = readFromExtensionsYml(extensionsPath.resolve("extensions.yml")).getExtensions();
             } catch (Exception e) {
```

Comment:
Generally, catching all exceptions can hide other bugs.

`readFromExtensionsYml` throws `IOException`. Is there an intentional reason we are being overly broad in catching all exceptions here?

Reply:
I thought I had changed that in my last PR, per your suggestion.  Looks like it got lost when I was rebasing and merging and I didn't catch it